zookeeper-jute/src/main/java/org/apache/jute/BinaryInputArchive.java

```diff
@@ -22,108 +22,124 @@
 import java.io.DataInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 
 /**
  *
  */
 public class BinaryInputArchive implements InputArchive {
-    static public final String UNREASONBLE_LENGTH= "Unreasonable length = ";
+
+    public static final String UNREASONBLE_LENGTH = "Unreasonable length = ";
+    public static final int MAX_BUFFER = Integer.getInteger("jute.maxbuffer", 0xfffff);
+
     private DataInput in;
-    
-    static public BinaryInputArchive getArchive(InputStream strm) {
+
+    public static BinaryInputArchive getArchive(InputStream strm) {
         return new BinaryInputArchive(new DataInputStream(strm));
     }
-    
-    static private class BinaryIndex implements Index {
+
+    private static class BinaryIndex implements Index {
         private int nelems;
+
         BinaryIndex(int nelems) {
             this.nelems = nelems;
         }
+
         public boolean done() {
             return (nelems <= 0);
         }
+
         public void incr() {
             nelems--;
         }
     }
-    /** Creates a new instance of BinaryInputArchive */
+
+    /**
+     * Creates a new instance of BinaryInputArchive.
+     */
     public BinaryInputArchive(DataInput in) {
         this.in = in;
     }
-    
+
     public byte readByte(String tag) throws IOException {
         return in.readByte();
     }
-    
+
     public boolean readBool(String tag) throws IOException {
         return in.readBoolean();
     }
-    
+
     public int readInt(String tag) throws IOException {
         return in.readInt();
     }
-    
+
     public long readLong(String tag) throws IOException {
         return in.readLong();
     }
-    
+
     public float readFloat(String tag) throws IOException {
         return in.readFloat();
     }
-    
+
     public double readDouble(String tag) throws IOException {
         return in.readDouble();
     }
-    
+
     public String readString(String tag) throws IOException {
-    	int len = in.readInt();
-    	if (len == -1) return null;
+        int len = in.readInt();
+        if (len == -1) {
+            return null;
+        }
         checkLength(len);
-    	byte b[] = new byte[len];
-    	in.readFully(b);
-    	return new String(b, "UTF8");
+        byte[] b = new byte[len];
+        in.readFully(b);
+        return new String(b, StandardCharsets.UTF_8);
     }
-    
-    static public final int maxBuffer = Integer.getInteger("jute.maxbuffer", 0xfffff);
```

Comment:
We can't rename a 'public static final' constant.
It is an API break we can create an issue to gather all the changes that we are not doing because we can't.
We have to keep code 100% compatible with the prev version
You should add a suppression (there is a way to add a directive in source code 'CHECKSTYLE OFF'

Reply:
Yes, I will update.